pkg/cluster/config_manager.go

```diff
@@ -0,0 +1,168 @@
+package cluster
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/yaml"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+type (
+	APIObjectGenerator func() APIObject
+	ParsedProcessor    func(*Config, ObjectLookup)
+	Validation         func(*Config) error
+	Defaulter          func(*Config)
+)
+
+type ConfigManager struct { // TODO: find a better name
+	entry *ConfigManagerEntry
+}
+
+func NewConfigManager() *ConfigManager {
+	return &ConfigManager{
+		entry: NewConfigManagerEntry(),
+	}
+}
+
+func (c *ConfigManager) Register(entries ...*ConfigManagerEntry) error {
+	return c.entry.Merge(entries...)
+}
+
+func (c *ConfigManager) RegisterMapping(kind string, generator APIObjectGenerator) error {
```

Comment:
I had a thought. If the intent is for consumers to model `ConfigManagerEntry`s as everything needed for mapping, defaulting and validating, why allow consumers to register things individually? 

It just feels odd to encourage that and then make the statement that they're individual concepts on the same construct that wants to model them as a single concept?

Reply:
How I see it, they are just two different options to do the same thing. `ConfigManagerEntry` exists only for convenience, not to enforce a particular pattern.
If you are going to register validations and defaults at the same time, you can use `ConfigManagerEntry`. But if you prefer, you can register them in separate calls.

Reply:
Ok. I think we could add something to `ConfigManagerEntry` docs to explicitly state its a way to conveniently model everything associated with a particular type and is semantically equivalent to performing individual registrations to avoid any confusion.

Reply:
Done